go-controller/pkg/ovn/pods.go

```diff
@@ -443,17 +443,30 @@ func (oc *Controller) addLogicalPort(pod *kapi.Pod) (err error) {
 	routingExternalGWs := oc.getRoutingExternalGWs(pod.Namespace)
 	routingPodGWs := oc.getRoutingPodGWs(pod.Namespace)
 
+	podHasGWRoute := false
 	// if we have any external or pod Gateways, add routes
-	if len(routingExternalGWs) > 0 || len(routingPodGWs) > 0 {
-		routingGWs := routingExternalGWs
-		for _, ipNets := range routingPodGWs {
-			routingGWs = append(routingGWs, ipNets...)
-		}
-		err = oc.addGWRoutesForPod(routingGWs, podIfAddrs, pod.Namespace, pod.Spec.NodeName)
+	if len(routingExternalGWs.gws) > 0 {
+		err = oc.addGWRoutesForPod(routingExternalGWs, podIfAddrs, pod.Namespace, pod.Spec.NodeName)
```

Comment:
is there anyway we can consolidate this and line 461 to a single call for addGWRoutesForPod? We were only calling it once before, and now we are calling it twice. It seems like an expensive function because we have to get a lock and then loop through a bunch of stuff again.

Reply:
Yup will change it somehow to accept the two gws. 
I missed the loop over podIfAddrs, makes sense to have it only once.